File: src/com/l2kt/gameserver/network/serverpackets/ExDuelUpdateUserInfo.java
package com.l2kt.gameserver.network.serverpackets;

import com.l2kt.gameserver.model.actor.instance.Player;

/**
 * Format: ch Sddddddddd
 * @author KenM
 */
public class ExDuelUpdateUserInfo extends L2GameServerPacket
{
	private final Player _activeChar;
	
	public ExDuelUpdateUserInfo(Player cha)
	{
		_activeChar = cha;
	}
	
	@Override
	protected void writeImpl()
	{
		writeC(0xfe);
		writeH(0x4f);
		writeS(_activeChar.getName());
		writeD(_activeChar.getObjectId());
		writeD(_activeChar.getClassId().getId());
		writeD(_activeChar.getLevel());
		writeD((int) _activeChar.getCurrentHp());
		writeD(_activeChar.getMaxHp());
		writeD((int) _activeChar.getCurrentMp());
		writeD(_activeChar.getMaxMp());
		writeD((int) _activeChar.getCurrentCp());
		writeD(_activeChar.getMaxCp());
	}
}

File: src/com/l2kt/gameserver/network/serverpackets/ExDuelUpdateUserInfo.kt
package com.l2kt.gameserver.network.serverpackets

import com.l2kt.gameserver.model.actor.instance.Player

/**
 * Format: ch Sddddddddd
 * @author KenM
 */
class ExDuelUpdateUserInfo(private val _activeChar: Player) : L2GameServerPacket() {

    override fun writeImpl() {
        writeC(0xfe)
        writeH(0x4f)
        writeS(_activeChar.name)
        writeD(_activeChar.objectId)
        writeD(_activeChar.classId.id)
        writeD(_activeChar.level)
        writeD(_activeChar.currentHp.toInt())
        writeD(_activeChar.maxHp)
        writeD(_activeChar.currentMp.toInt())
        writeD(_activeChar.maxMp)
        writeD(_activeChar.currentCp.toInt())
        writeD(_activeChar.maxCp)
    }
}